simple-dns/src/dns/rdata/ipseckey.rs

```diff
@@ -0,0 +1,191 @@
+use crate::{dns::WireFormat, Name};
+use std::{borrow::Cow, net::{Ipv4Addr, Ipv6Addr}};
+
+use super::RR;
+
+/// IPSECKEY record type stores information about IPsec key material
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub struct IPSECKEY<'a> {
+    /// Precedence for this record, lower values are preferred
+    pub precedence: u8,
+    /// Type of gateway (0=none, 1=IPv4, 2=IPv6, 3=domain name)
+    gateway_type: u8,
```

Comment:
Since this field only exists to determine the value of the `gateway` field, which you already implemented an enum for, I think it is best to remove this field and just use the enum to generate this value when writing the wire format.

Otherwise it would be necessary to add a `new` function to this struct, since the private `gateway_type` field won't allow the struct to be constructed by code outside of this crate